zerver/tests/test_message_fetch.py

```diff
@@ -1197,6 +1197,45 @@ def test_successful_get_messages(self) -> None:
             ),
         )
 
+    def test_unauthenticated_get_messages_without_web_public(self) -> None:
+        """
+        An unauthenticated call to GET /json/messages with valid parameters
+        returns a 401.
+        """
+        post_params = {
+            "anchor": 1,
+            "num_before": 1,
+            "num_after": 1
+        }  # type: Dict[str, Union[str, int]]
+        result = self.client_get("/json/messages", dict(post_params))
+        self.assert_json_error(result, 'Not logged in: API authentication or user session required',
+                               status_code=401)
+
+    def test_unauthenticated_get_messages_with_web_public(self) -> None:
+        """
+        An unauthenticated call to GET /json/messages with valid parameters
+        including `is_web_public_query` returns a list of messages.
+        """
+        post_params = {
+            "anchor": 1,
+            "num_before": 1,
+            "num_after": 1,
+            "is_web_public_query": 'true',
+        }  # type: Dict[str, Union[str, int]]
+        result = self.client_get("/json/messages", dict(post_params))
+        self.assert_json_success(result)
+
+        post_params = {
+            "anchor": 1,
+            "num_before": 1,
+            "num_after": 1,
+            "narrow": ujson.dumps([dict(operator='is', operand="private")]),
+            "is_web_public_query": 'true',
+        }
+        result = self.client_get("/json/messages", dict(post_params))
+        self.assert_json_error(result, "Not logged in: API authentication or user session required",
+                               status_code=401)
+
```

Comment:
We're going to want to extend this test with a bit more test setup to check that the response doesn't get any messages to public streams -- just those to web-public streams.

Reply:
Done!